File: ReserseString/reverseString.java
/**
 * Created by XiaochengWen on 16/9/24.
 */
public class reverseString {
    public String reverseString(String s) {
        char[] chars = s.toCharArray();
        int length = chars.length;
        int n = length - 1;
        char t;
        for (int i = n >> 1; i >= 0; i--) {
            int k = n - i;
            t = chars[i];
            chars[i] = chars[k];
            chars[k] = t;
        }
        return new String(chars);
    }
}

File: ReserseString/Reverse String.py
__author__ = 'XiaochengWen'
class Solution(object):
    def reverseString(self, s):
        a = list(s)
        i=0
        while 2*i<len(s)-1:
            a[i],a[len(s)-i-1] = a[len(s)-i-1],a[i]
            i+=1
        return ''.join(a)
a = Solution()
print(a.reverseString("djasf"))
